src/ip/QosConfig.cc

```diff
@@ -531,15 +531,22 @@ Ip::Qos::setSockTos(const int fd, tos_t tos, int type)
 {
     // Bug 3731: FreeBSD produces 'invalid option'
     // unless we pass it a 32-bit variable storing 8-bits of data.
-    // NP: it is documented as 'int' for all systems, even those like Linux which accept 8-bit char
-    //     so we convert to a int before setting.
+    // NOTE: setsockopt(.., IP_TOS, ...) 'optval' parameter
+    //       is documented as 'int' for all systems
+    //       despite what the function parameter type is.
+    //       So we convert to int before setting.
```

Comment:
Please drop the `NOTE:` prefix. Every comment is a NOTE by default, and there is no good reason to maintain the associated indentation. I would also simplify to avoid using the first "parameter" word, to clarify what type we are talking about. Together with the other change request suggestion, this comment will become very simple:

```suggestion
    // setsockopt(.., IP_TOS, ...) option value is documented to be 'int' for all systems
```

or even

```suggestion
    // IP_TOS option value is documented to be 'int' for all systems
```

We could add a comment about casting to the place where we actually cast, but I would not do that because Squid has dozens of such casts and some of them are already documented. There is no need to document this particular cast in this particular place IMO[^1]. There is nothing special about this particular cast, and it is virtually unrelated to the type of the option value (technically, the explicit cast could be avoided if the type was `char`, but the reader of this code does not really need to know that fact).

[^1]: especially since we should eventually migrate to a cast-free wrapper as discussed in the other change request